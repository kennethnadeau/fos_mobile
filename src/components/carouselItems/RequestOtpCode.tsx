import React, { FC } from "react";
import RoundedButton from "@fos/components/RoundedButton";
import TextInput from "@fos/components/TextInput";
import { useTranslation, Trans } from "react-i18next";
import { StyleSheet, View, Keyboard } from "react-native";
import { Text } from "react-native-elements";
import { ms, s, vs } from "react-native-size-matters";
import CarouselItemContainer from "./CarouselItemContainer";
import FooterText from "./FooterText";

export type RequestCodeProps = {
  mobileNumber: string;
  onMobileNumberChangeText: (value: string) => void;
  countryCode: string;
  onCountryCodeChange: (value: string) => void;
  onMobileNumberClear: () => void;
  onRequestCodePress: () => void;
  loading?: boolean;
};

const RequestOtpCode: FC<RequestCodeProps> = ({
  countryCode,
  onCountryCodeChange,
  mobileNumber,
  onMobileNumberChangeText,
  onMobileNumberClear,
  onRequestCodePress,
  loading,
}) => {
  const { t } = useTranslation("carouselItems");

  const enterMobileNumberText = t("Enter Mobile Number");
  const yourNumberText = t("Your Number");
  const requestCodeText = t("Request Code");

  const onSubmit = () => {
    onRequestCodePress();
    Keyboard.dismiss();
  };

  return (
    <CarouselItemContainer
      containerProps={{
        accessible: true,
        accessibilityLabel: enterMobileNumberText,
      }}
      header={enterMobileNumberText}>
      <View style={styles.inputContainer}>
        <TextInput
          accessibilityLabel={t("Country code")}
          accessible
          containerStyle={styles.countryCodeContainer}
          hideClearIcon
          inputContainerStyle={styles.countryCodeInputContainer}
          keyboardType="phone-pad"
          onChangeText={onCountryCodeChange}
          placeholder="US +1"
          value={countryCode}
        />
        <TextInput
          accessibilityLabel={yourNumberText}
          accessible
          containerStyle={styles.mobileNumberContainer}
          inputContainerStyle={styles.mobileNumberInputContainer}
          keyboardType="phone-pad"
          onChangeText={onMobileNumberChangeText}
          onClear={onMobileNumberClear}
          onSubmitEditing={onSubmit}
          placeholder={yourNumberText}
          value={mobileNumber}
        />
      </View>
      <RoundedButton
        accessibilityLabel={requestCodeText}
        accessible
        containerStyle={styles.btnRequestCode}
        disabled={!mobileNumber || loading}
        loading={loading}
        onPress={onSubmit}
        size="large"
        title={requestCodeText}
      />
      <FooterText style={styles.terms}>
        <Trans components={{ Text: <Text /> }}>Agree to Terms of Use</Trans>
      </FooterText>

      <FooterText style={styles.disclaimer}>{t("OTP Disclaimer")}</FooterText>
    </CarouselItemContainer>
  );
};

const styles = StyleSheet.create({
  btnRequestCode: {
    width: "100%",
  },

  countryCodeContainer: {
    paddingHorizontal: 0,
    paddingRight: ms(10),
    width: "35%",
  },
  countryCodeInputContainer: {
    padding: s(8),
  },
  disclaimer: {
    marginVertical: vs(25),
    textAlign: "center",
  },
  inputContainer: {
    flexDirection: "row",
    justifyContent: "space-between",
    marginVertical: vs(10),
  },
  mobileNumberContainer: {
    paddingHorizontal: 0,
    width: "65%",
  },
  mobileNumberInputContainer: {
    padding: s(4),
  },

  terms: {
    marginTop: vs(30),
    textAlign: "center",
  },
});

export default RequestOtpCode;
