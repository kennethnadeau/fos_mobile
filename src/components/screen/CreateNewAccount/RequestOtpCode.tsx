import React, {FC} from 'react';
import RoundedButton from '@fos/components/RoundedButton';
import TextInput from '@fos/components/TextInput';
import {useTranslation} from 'react-i18next';
import {StyleSheet, View, Keyboard} from 'react-native';
import {Text} from 'react-native-elements';
import {ms, s, vs} from 'react-native-size-matters';
import CarouselItemContainer from './CarouselItemContainer';
import FooterText from './FooterText';

export type RequestCodeProps = {
  mobileNumber: string;
  onMobileNumberChangeText: (value: string) => void;
  countryCode: string;
  onCountryCodeChange: (value: string) => void;
  onMobileNumberClear: () => void;
  onRequestCodePress: () => void;
  loading?: boolean;
};

const RequestOtpCode: FC<RequestCodeProps> = ({
  countryCode,
  onCountryCodeChange,
  mobileNumber,
  onMobileNumberChangeText,
  onMobileNumberClear,
  onRequestCodePress,
  loading,
}) => {
  const {t: translate} = useTranslation();

  const t = (path: string) => translate(`screens.createNewAccount.${path}`);

  const onSubmit = () => {
    onRequestCodePress();
    Keyboard.dismiss();
  };

  return (
    <CarouselItemContainer
      containerProps={{
        accessible: true,
        accessibilityLabel: t('requestOtpCodeHeader'),
      }}
      header={t('requestOtpCodeHeader')}>
      <View style={styles.inputContainer}>
        <TextInput
          accessibilityLabel={t('requestOtpCodeInput.countryCode')}
          accessible
          containerStyle={styles.countryCodeContainer}
          hideClearIcon
          inputContainerStyle={styles.countryCodeInputContainer}
          keyboardType="phone-pad"
          onChangeText={onCountryCodeChange}
          placeholder="US +1"
          value={countryCode}
        />
        <TextInput
          accessibilityLabel={t('requestOtpCodeInput.mobileNumber')}
          accessible
          containerStyle={styles.mobileNumberContainer}
          inputContainerStyle={styles.mobileNumberInputContainer}
          keyboardType="phone-pad"
          onChangeText={onMobileNumberChangeText}
          onClear={onMobileNumberClear}
          onSubmitEditing={onSubmit}
          placeholder={t('requestOtpCodeInput.mobileNumber')}
          value={mobileNumber}
        />
      </View>
      <RoundedButton
        accessibilityLabel={t('requestOtpCodeButton.requestCode')}
        accessible
        containerStyle={styles.btnRequestCode}
        disabled={!mobileNumber || loading}
        loading={loading}
        onPress={onSubmit}
        size="large"
        title={t('requestOtpCodeButton.requestCode')}
      />
      <FooterText style={styles.terms}>
        {t('requestOtpCodeFooter.terms')}
      </FooterText>
      <Text>{t('requestOtpCodeFooter.termsOfUse')}.</Text>

      <FooterText style={styles.disclaimer}>
        {t('requestOtpCodeFooter.disclaimer')}
      </FooterText>
    </CarouselItemContainer>
  );
};

const styles = StyleSheet.create({
  btnRequestCode: {
    width: '100%',
  },

  countryCodeContainer: {
    paddingHorizontal: 0,
    paddingRight: ms(10),
    width: '35%',
  },
  countryCodeInputContainer: {
    padding: s(8),
  },
  disclaimer: {
    marginVertical: vs(25),
    textAlign: 'center',
  },
  inputContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginVertical: vs(10),
  },
  mobileNumberContainer: {
    paddingHorizontal: 0,
    width: '65%',
  },
  mobileNumberInputContainer: {
    padding: s(4),
  },

  terms: {
    marginTop: vs(30),
    textAlign: 'center',
  },
});

export default RequestOtpCode;
