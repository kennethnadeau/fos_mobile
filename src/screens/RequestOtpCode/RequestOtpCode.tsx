import React, { useCallback, useLayoutEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";

import { ScreenWrapper } from "@fos/components/ScreenWrapper";
import { SCREENS } from "@fos/constants";
import { apiService } from "@fos/shared";
import { setToastMessage } from "@fos/redux/slices/toastSlice";
import RequestOtpCode from "@fos/components/Account/RequestOtpCode";
import {
  setCountryCode,
  setMobileNumber,
  clearOtpForm,
} from "redux/reducers/otpReducer";
import { selectOtpState } from "redux/selectors/otpSelectors";
import { ScreenFC } from "react-native-navigation-register-screens";
import { goToVerifyScreen } from "helpers/navigation";
import {
  setCurrentScreen,
  updatePaginationActiveDotIndex,
} from "redux/slices/navigationSlice";
import { getDotIndex } from "../helpers";

const { otp } = apiService;

// TODO: figure out type here
const RequestOtpCodeScreen: ScreenFC<any> = (props) => {
  const { login, componentId } = props;
  const dispatch = useDispatch();
  const otpState = useSelector(selectOtpState);

  const [otpRequestStatus, setOtpRequestStatus] = useState<
    "sending" | "sent" | "idle"
  >("idle");

  const { countryCode, mobileNumber } = otpState;
  const updateCountryCode = (code: string) => {
    dispatch(setCountryCode(code));
  };
  const clearMobileNumber = () => {
    dispatch(setMobileNumber(""));
  };
  const updateMobileNumber = (mobile: string) => {
    dispatch(setMobileNumber(mobile));
  };

  useLayoutEffect(() => {
    dispatch(clearOtpForm());
    dispatch(setCurrentScreen(SCREENS.REQUEST_CODE));
    const dotIndex = getDotIndex(SCREENS.REQUEST_CODE, login);
    dispatch(updatePaginationActiveDotIndex(dotIndex));
  }, [dispatch, login]);

  const formatPhoneNumber = useCallback(
    () => `${countryCode.replace("+", "")}${mobileNumber}`,
    [countryCode, mobileNumber],
  );

  const sendOtpCode = useCallback(() => {
    const postRequestMethod = login
      ? otp.postOtpAuthenticate
      : otp.postOtpRegistration;

    return postRequestMethod({ phone: formatPhoneNumber() });
  }, [formatPhoneNumber, login]);

  const handleOtpCodeRequest = async () => {
    setOtpRequestStatus("sending");
    try {
      await sendOtpCode();
      setOtpRequestStatus("sent");
      goToVerifyScreen(componentId, login);
    } catch (e) {
      dispatch(setToastMessage("Whoops! Something went wrong!"));
    } finally {
      setOtpRequestStatus("idle");
    }
  };

  return (
    <ScreenWrapper login={login}>
      <RequestOtpCode
        {...{ countryCode, mobileNumber }}
        loading={otpRequestStatus === "sending"}
        onCountryCodeChange={updateCountryCode}
        onMobileNumberChangeText={updateMobileNumber}
        onMobileNumberClear={clearMobileNumber}
        onRequestCodePress={handleOtpCodeRequest}
      />
    </ScreenWrapper>
  );
};

RequestOtpCodeScreen.screenName = SCREENS.REQUEST_CODE;
RequestOtpCodeScreen.options = {
  topBar: {
    visible: false,
    rightButtons: [
      {
        id: SCREENS.COMPONENTS.TOP_BAR.QUESTION_MARK,
        component: {
          name: SCREENS.COMPONENTS.TOP_BAR.QUESTION_MARK,
        },
      },
    ],
  },
};

export default RequestOtpCodeScreen;
