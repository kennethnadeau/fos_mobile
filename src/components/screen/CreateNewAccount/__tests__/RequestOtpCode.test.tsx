import React from 'react';
import RequestOtpCode, {RequestCodeProps} from '../RequestOtpCode';
import {render, RenderAPI} from '@fos/utils/testing';
import enUS from '@fos/locale/languages/en-US';

const localization = enUS.translation.screens.createNewAccount;

describe('<RequestOtpCode />', () => {
  const props: RequestCodeProps = {
    onRequestCodePress: jest.fn(),
    countryCode: '',
    mobileNumber: '',
    onCountryCodeChange: jest.fn(),
    onMobileNumberChangeText: jest.fn(),
    onMobileNumberClear: jest.fn(),
  };

  let requestCodeComponent: RenderAPI;

  beforeEach(() => {
    requestCodeComponent = render(<RequestOtpCode {...props} />);
  });

  it('should have carousel item description', () => {
    const actual = requestCodeComponent.queryByA11yLabel(
      localization.requestOtpCodeHeader,
    );

    expect(actual).not.toBeNull();
  });

  it('should show header', () => {
    const actual = requestCodeComponent.queryByText(
      localization.requestOtpCodeHeader,
    );
    expect(actual).not.toBeNull();
  });

  it('should show country code input', () => {
    const actual = requestCodeComponent.queryByA11yLabel(
      localization.requestOtpCodeInput.countryCode,
    );

    expect(actual).not.toBeNull();
  });

  it('should show mobile number input', () => {
    const actual = requestCodeComponent.queryByPlaceholder(
      localization.requestOtpCodeInput.mobileNumber,
    );

    expect(actual).not.toBeNull();
  });

  it('should show request code button', () => {
    const actual = requestCodeComponent.queryByA11yLabel(
      localization.requestOtpCodeButton.requestCode,
    );
    expect(actual).not.toBeNull();
  });

  it('should match snapshot', () => {
    expect(requestCodeComponent.toJSON()).toMatchSnapshot();
  });
});
