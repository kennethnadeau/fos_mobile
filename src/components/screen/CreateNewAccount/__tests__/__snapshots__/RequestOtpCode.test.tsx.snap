// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestOtpCode /> should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityLabel="Enter Mobile Number"
    accessible={true}
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "paddingHorizontal": 85.71428571428571,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 27.5,
          "marginVertical": 39.23529411764706,
        }
      }
      theme={
        Object {
          "Button": Object {
            "buttonStyle": Array [
              Object {
                "backgroundColor": "#4CA1D6",
              },
            ],
            "titleStyle": Array [
              Object {
                "color": "white",
                "fontSize": 34.285714285714285,
              },
            ],
          },
          "Icon": Object {
            "color": "#2089dc",
          },
          "Input": Object {
            "inputContainerStyle": Object {
              "alignItems": "center",
              "borderBottomWidth": 1.4999999999999998,
              "borderColor": "#fff",
              "borderRadius": 64.28571428571428,
              "borderWidth": 1.4999999999999998,
              "overflow": "hidden",
              "padding": 9.428571428571429,
            },
            "inputStyle": Object {
              "alignItems": "center",
              "color": "#fff",
              "padding": 9.428571428571429,
            },
            "placeholderTextColor": "#7E8395",
          },
          "Text": Object {
            "style": Object {
              "color": "white",
              "fontSize": 34.285714285714285,
            },
          },
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      Enter Mobile Number
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 19.61764705882353,
        }
      }
    >
      <View
        style={
          Object {
            "paddingHorizontal": 0,
            "paddingRight": 15.714285714285714,
            "width": "35%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1.4999999999999998,
              "borderColor": "#fff",
              "borderRadius": 64.28571428571428,
              "borderWidth": 1.4999999999999998,
              "flexDirection": "row",
              "overflow": "hidden",
              "padding": 17.142857142857142,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            accessibilityLabel="Country code"
            accessible={true}
            allowFontScaling={true}
            editable={true}
            keyboardType="phone-pad"
            onChangeText={[MockFunction]}
            placeholder="US +1"
            placeholderTextColor="#7E8395"
            rejectResponderTermination={true}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "color": "#fff",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
                "padding": 9.428571428571429,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "margin": 5,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "paddingHorizontal": 0,
            "width": "65%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1.4999999999999998,
              "borderColor": "#fff",
              "borderRadius": 64.28571428571428,
              "borderWidth": 1.4999999999999998,
              "flexDirection": "row",
              "overflow": "hidden",
              "padding": 8.571428571428571,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            accessibilityLabel="Your Number"
            accessible={true}
            allowFontScaling={true}
            editable={true}
            keyboardType="phone-pad"
            onChangeText={[MockFunction]}
            onSubmitEditing={[Function]}
            placeholder="Your Number"
            placeholderTextColor="#7E8395"
            rejectResponderTermination={true}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "color": "#fff",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
                "padding": 9.428571428571429,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "marginVertical": 4,
                "paddingRight": 4,
              }
            }
          >
            <View
              style={
                Object {
                  "overflow": "hidden",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "rgba(0,0,0,0)",
                          "fontSize": 24,
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Object {
                          "fontFamily": "FontAwesome",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                    testID="iconIcon"
                  >
                    ÔÅó
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "margin": 5,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
          "width": "100%",
        }
      }
    >
      <View
        accessibilityLabel="Request Code"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": true,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "hsl(208, 8%, 90%)",
              "borderColor": "#2089dc",
              "borderRadius": 85.71428571428571,
              "borderWidth": 0,
              "flexDirection": "row",
              "height": 107.14285714285714,
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": Object {
                  "color": Array [
                    208,
                    8,
                    63,
                  ],
                  "model": "hsl",
                  "valpha": 1,
                },
                "fontSize": 34.285714285714285,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Request Code
          </Text>
        </View>
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#7E8395",
          "fontSize": 34.285714285714285,
          "marginTop": 58.85294117647059,
          "textAlign": "center",
        }
      }
      theme={
        Object {
          "Button": Object {
            "buttonStyle": Array [
              Object {
                "backgroundColor": "#4CA1D6",
              },
            ],
            "titleStyle": Array [
              Object {
                "color": "white",
                "fontSize": 34.285714285714285,
              },
            ],
          },
          "Icon": Object {
            "color": "#2089dc",
          },
          "Input": Object {
            "inputContainerStyle": Object {
              "alignItems": "center",
              "borderBottomWidth": 1.4999999999999998,
              "borderColor": "#fff",
              "borderRadius": 64.28571428571428,
              "borderWidth": 1.4999999999999998,
              "overflow": "hidden",
              "padding": 9.428571428571429,
            },
            "inputStyle": Object {
              "alignItems": "center",
              "color": "#fff",
              "padding": 9.428571428571429,
            },
            "placeholderTextColor": "#7E8395",
          },
          "Text": Object {
            "style": Object {
              "color": "white",
              "fontSize": 34.285714285714285,
            },
          },
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      Agree to Terms of Use
    </Text>
    <Text
      style={
        Object {
          "color": "#7E8395",
          "fontSize": 34.285714285714285,
          "marginVertical": 49.04411764705882,
          "textAlign": "center",
        }
      }
      theme={
        Object {
          "Button": Object {
            "buttonStyle": Array [
              Object {
                "backgroundColor": "#4CA1D6",
              },
            ],
            "titleStyle": Array [
              Object {
                "color": "white",
                "fontSize": 34.285714285714285,
              },
            ],
          },
          "Icon": Object {
            "color": "#2089dc",
          },
          "Input": Object {
            "inputContainerStyle": Object {
              "alignItems": "center",
              "borderBottomWidth": 1.4999999999999998,
              "borderColor": "#fff",
              "borderRadius": 64.28571428571428,
              "borderWidth": 1.4999999999999998,
              "overflow": "hidden",
              "padding": 9.428571428571429,
            },
            "inputStyle": Object {
              "alignItems": "center",
              "color": "#fff",
              "padding": 9.428571428571429,
            },
            "placeholderTextColor": "#7E8395",
          },
          "Text": Object {
            "style": Object {
              "color": "white",
              "fontSize": 34.285714285714285,
            },
          },
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      OTP Disclaimer
    </Text>
  </View>
</RNCSafeAreaProvider>
`;
